src/amplitude/event.py

```diff
@@ -587,31 +979,103 @@ def __init__(self, price: float,
         self.revenue: Optional[float] = revenue
 
     def set_receipt(self, receipt: str, receipt_signature: str):
+        """Set the receipt and signature
+
+        Args:
+            receipt (str): The receipt identifier
+            receipt_signature (str): The receipt signature
+
+        Returns:
+            the Revenue instance itself
+        """
         self.receipt = receipt
         self.receipt_sig = receipt_signature
         return self
 
     def is_valid(self):
+        """Check if a revenue instance is a valid one
+
+        Returns:
+            True if price is a float number, False otherwise
+        """
         return isinstance(self.price, float)
 
     def to_revenue_event(self):
+        """Create and return a RevenueEvent instance, set revenue information as event_properties.
+
+        Returns:
+          A RevenueEvent object
+        """
         return RevenueEvent(event_properties=self.get_event_properties())
 
     def get_event_properties(self):
+        """Return a dictionary of revenue instance data used as event_properties of RevenueEvent
+
+        Returns:
+          A dict object
+        """
         event_properties = {}
         if self.properties:
-            event_properties = self.properties.copy()
+            event_properties = copy.deepcopy(self.properties)
         event_properties.update({constants.REVENUE_PRODUCT_ID: self.product_id,
```

Comment:
Why do we need to update these values after the deepcopy?

Reply:
properties are additional event properties besides revenue related ones.